src/biogeochem/DUSTMod.F90

```diff
@@ -356,86 +583,201 @@ subroutine DustEmission (bounds, &
+
+               ! account for bare soil fraction, frozen soil fraction, and apply global tuning parameter (Kok et al. 2014)
+               flx_mss_vrt_dst_ttl(p) = flx_mss_vrt_dst_ttl(p) * lnd_frc_mbl(p) * C_tune * liqfrac
+               !########################################################################################
             end if
 
-            ! the following comes from subr. flx_mss_hrz_slt_ttl_Whi79_get
-            ! purpose: compute vertically integrated streamwise mass flux of particles
+            !############## Danny M. Leung added the intermittency calculation #################################
+            ! subsection for intermittency factor calculation (only used by Leung's scheme, not Zender's scheme)
+            ! Leung et al. (2023) uses the Comola et al. (2019) intermittency scheme for the calculation of intermittent dust emissions.
+            ! This part takes care of the sub-timestep, high-frequency (< 1 minute period) turblent wind fluctuations occuring at the planetary boundary layer (PBL) near surface. Subtimestep wind gusts and episodes are important for generating emissions in marginal dust source regions, such as semiarid areas and high-latitude polar deserts.
+            ! 2 Dec 2021: assume no buoyancy contribution to the wind fluctuation (u_sd_slt), so no obul(p) is needed. It is shown to be important for the wind fluctuations contribute little to the intermittency factor. We might add this back in the future revisions.
+            ! 20 Feb 2024: dmleung notes that dmleung may revise Comola's scheme in the future to improve Comola's formulation of the statistical parameterization.
 
-            if (wnd_frc_slt > wnd_frc_thr_slt) then
-               wnd_frc_rat = wnd_frc_thr_slt / wnd_frc_slt
-               flx_mss_hrz_slt_ttl = cst_slt * forc_rho(c) * (wnd_frc_slt**3.0_r8) * &
-                    (1.0_r8 - wnd_frc_rat) * (1.0_r8 + wnd_frc_rat) * (1.0_r8 + wnd_frc_rat) / grav
+            ! mean lowpass-filtered wind speed at hgt_sal = 0.1 m saltation height (assuming aerodynamic roughness length z0a_glob = 1e-4 m globally for ease; also assuming neutral condition)
+            u_mean_slt(p) = (wnd_frc_slt/k) * log(hgt_sal / z0a_glob)  ! translating from ustar (velocity scale) to actual wind
 
-               ! the following loop originates from subr. dst_mbl
-               ! purpose: apply land sfc and veg limitations and global tuning factor
-               ! slevis: multiply flx_mss_hrz_slt_ttl by liqfrac to incude the effect 
-               ! of frozen soil
+            !stblty(p) = 0_r8   ! -dmleung 2 Dec 2021: use stability = 0 for now, assuming no buoyancy contribution. Might uncomment the above lines in future revisions.
```

Comment:
@dmleung s setting stblty to zero still a something useful to do? And useful to others working on dust? My preference would be to delete the commented out line if it's only something you might do for testing.

Reply:
I recently included an expression for the stability when I incorporate the similarity theory into the dust scheme, so I think this `stblty(p) = 0_r8` is no longer needed. Please feel free to remove it!

Reply:
Still should be done.